pkg/awsManager/vpcVpnManager.go

```diff
@@ -0,0 +1,629 @@
+		}
+		//detach VPN gateway
+		err = DetachVpnGateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//cleaning network ACL
+		err = DeleteNetworkAcl(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//now cleaning route tables
+		err = DeleteRouteTables(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// now cleaning subnets associated with that vpc id
+		err = DeleteSubnetsForVPC(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//now cleaning security groups
+		err = DeleteSecurityGroups(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+
+		_, err = client.DeleteVpc(&ec2.DeleteVpcInput{VpcId: vpcId})
+		if err != nil {
+			errFlag = true
+			fmt.Println("ERROR deleting VPC :", *vpcId)
+			fmt.Println(err)
+		}
+	}
+
+	if errFlag == true {
+		return errors.New("ERROR")
+	} else {
+		fmt.Println("VPC's deleted successfully for this region  ")
+		return nil
+	}
+
+}
+func DeleteELB(client clientpkg.Client, vpcID *string) error {
+
+	errFlag := false
+	var marker *string
+	for {
+		elbList, err := client.DescribeLoadBalancers(&elb.DescribeLoadBalancersInput{Marker: marker})
+		if err != nil {
+			fmt.Println("ERROR: cant list ELB")
+			fmt.Println(err)
+			errFlag = true
+		}
+
+		for _, elasticLoadBalancer := range elbList.LoadBalancerDescriptions {
+			if *elasticLoadBalancer.VPCId == *vpcID {
+				_, err := client.DeleteLoadBalancer(&elb.DeleteLoadBalancerInput{LoadBalancerName: elasticLoadBalancer.LoadBalancerName})
+				if err != nil {
+					fmt.Println("ERROR: cant delete ELB", *elasticLoadBalancer.LoadBalancerName)
+					fmt.Println(err)
+					errFlag = true
+				}
+			}
+		}
+
+		if elbList.NextMarker != nil {
+			marker = elbList.NextMarker
+		} else {
+			break
+		}
+
+	}
+
+	if errFlag == true {
+		fmt.Println("ERROR : Some ELB not removed. Refer to logs above.")
+		return errors.New("ERROR")
+	}
+
+	fmt.Println("All ELB were removed successfully for this region")
+	return nil
+
+}
+
+func DeleteNatgateway(client clientpkg.Client, vpcID *string) error {
+
+	errFlag := false
+	var token *string
+	for {
+		natGatewayList, err := client.DescribeNatGateways(&ec2.DescribeNatGatewaysInput{NextToken: token})
+		if err != nil {
+			fmt.Println("ERROR: cant list NAT GATEWAY")
+			fmt.Println(err)
+			errFlag = true
+		}
+
+		for _, natGateway := range natGatewayList.NatGateways {
+			if *natGateway.VpcId == *vpcID {
+				_, err := client.DeleteNatGateway(&ec2.DeleteNatGatewayInput{NatGatewayId: natGateway.NatGatewayId})
+				if err != nil {
+					fmt.Println("ERROR: cant delete NAT GATEWAY ID", *natGateway.NatGatewayId)
+					fmt.Println(err)
+					errFlag = true
+				}
+
+			}
+		}
+
+		if natGatewayList.NextToken != nil {
+			token = natGatewayList.NextToken
+		} else {
+			break
+		}
+	}
+
+	if errFlag == true {
+		fmt.Println("ERROR : Not all NAT gateway ID have been deleted. Refer to Logs above")
+		return errors.New("ERROR")
+	}
+
+	fmt.Println("NAT GATEWAY ID removed")
```

Comment:
Did you mean to put the ID in this print?

Reply:
fixed the log messages . the undeleted objects are printed before